src/main/java/org/folio/rest/impl/VendorHelper.java

```diff
@@ -100,12 +101,14 @@ public VendorHelper(Map<String, String> okapiHeaders, Context ctx, String lang)
           .collect(toList());
         ids.stream()
           .filter(id -> !vendorsIds.contains(id))
-          .forEach(id -> errors.add(POL_ACCESS_PROVIDER_NOT_FOUND.toError().withAdditionalProperty(ID, id)));
+          .forEach(id -> errors.add(createErrorWithId(POL_ACCESS_PROVIDER_NOT_FOUND, id)));
         return handleAndReturnErrors(errors);
       }).thenAccept(future::complete)
         .exceptionally(t -> {
-          logger.error("Failed to validate access provider's status", t);
-          future.completeExceptionally(t);
+          Throwable cause = t.getCause();
+          logger.error("Failed to validate access provider's status", cause);
+          errors.add(VENDOR_ISSUE.toError().withMessage(cause.getMessage()));
```

Comment:
I wonder if it's worth being more specific about vendor issues... e.g. having separate codes for the various issues... Or, at least passing the same message you log above instead of the one from the cause?  

I say this not knowing what the cause message looks like in this case... So feel free to push back

Reply:
There are really 2 ways to handle such unexpected errors:
1. Hide all the details and always use static/generic message.
In this case the only source to understand the cause of the problem is back-end application logs.
2. Use static/generic error code but provide details of the issue as error message. 
In this case any client of the API can contact back-end developers (of course by jira ticket) with useful details describing the problem

The UI application is not going to use error message but just display to end user localized message which is based on error code. So most of the consumers of the UI application will never know about real cause...
My personal preference as of back-end developer to be able to see details of the issue just using Browser's console and then use logs to have more details if required.

For example, the error for the case this PR is created for looks like following:
```json
{
  "message" : "Cannot deserialize instance of `java.lang.String` out of START_OBJECT token\n at [Source: UNKNOWN; line: -1, column: -1] (through reference chain: org.folio.rest.acq.model.Vendor[\"contacts\"]->java.util.ArrayList[0])",
  "code" : "vendorIssue",
  "parameters" : [ ]
}
```

Reply:
I like the idea of having the additional "cause" field.  I didn't realize that was an option.

> The UI application is not going to use error message but just display to end user localized message which is based on error code. So most of the consumers of the UI application will never know about real cause...

This is exactly why I made the suggestion of using discrete codes instead of a general vendorIssue code.  It might be helpful if the UI could provide the user with a specific localized message which indicates what the problem was instead of just that there was a problem.

Reply:
To be honest I do not know what error code we might return because this is exceptionally section which means something unexpected happened...
By the way I changed the handling a bit and here is the output of the sample response (from the unit test) for the unexpected error verifying order vendor and PO Line's acceess provider:
```json
{
  "errors": [
    {
      "message": "Order cannot be updated due to issues related to associated vendor/access provider",
      "code": "vendorIssue",
      "parameters": [
        {
          "key": "id",
          "value": "5a34ae0e-5a11-4337-be95-1a20cfdc3161"
        }
      ],
      "cause": "Unrecognized field \"contactus\" (class org.folio.rest.acq.model.Vendor), not marked as ignorable (36 known properties: \"vendor_types\", \"vendor_currencies\", \"licensor\", \"vendor_status\", \"name\", \"contacts\", \"emails\", \"language\", \"interfaces\", \"id\", \"governmental\", \"changelogs\", \"addresses\", \"urls\", \"accounts\", \"subscription_interval\", \"agreements\", \"edi\", \"access_provider\", \"code\", \"tax_percentage\", \"tax_id\", \"phone_numbers\", \"payment_method\", \"description\", \"aliases\", \"discount_percent\", \"erp_code\", \"liable_for_vat\", \"expected_receipt_interval\", \"material_supplier\", \"san_code\", \"claiming_interval\", \"expected_activation_interval\", \"expected_invoice_interval\", \"renewal_activation_interval\"])\n at [Source: UNKNOWN; line: -1, column: -1] (through reference chain: org.folio.rest.acq.model.Vendor[\"contactus\"])"
    },
    {
      "message": "Order cannot be updated due to issues related to associated vendor/access provider",
      "code": "vendorIssue",
      "parameters": [],
      "cause": "Unrecognized field \"contactus\" (class org.folio.rest.acq.model.Vendor), not marked as ignorable (36 known properties: \"vendor_types\", \"vendor_currencies\", \"licensor\", \"vendor_status\", \"name\", \"contacts\", \"emails\", \"language\", \"interfaces\", \"id\", \"governmental\", \"changelogs\", \"addresses\", \"urls\", \"accounts\", \"subscription_interval\", \"agreements\", \"edi\", \"access_provider\", \"code\", \"tax_percentage\", \"tax_id\", \"phone_numbers\", \"payment_method\", \"description\", \"aliases\", \"discount_percent\", \"erp_code\", \"liable_for_vat\", \"expected_receipt_interval\", \"material_supplier\", \"san_code\", \"claiming_interval\", \"expected_activation_interval\", \"expected_invoice_interval\", \"renewal_activation_interval\"])\n at [Source: UNKNOWN; line: -1, column: -1] (through reference chain: org.folio.rest.acq.model.Vendor[\"contactus\"])"
    }
  ],
  "total_records": 2
}
```